sklearn/decomposition/_nmf.py

```diff
@@ -1308,23 +1293,94 @@ def fit_transform(self, X, y=None, W=None, H=None):
                                 dtype=[np.float64, np.float32])
 
         with config_context(assume_finite=True):
-            W, H, n_iter_ = non_negative_factorization(
-                X=X, W=W, H=H, n_components=self.n_components, init=self.init,
-                update_H=True, solver=self.solver, beta_loss=self.beta_loss,
-                tol=self.tol, max_iter=self.max_iter, alpha=self.alpha,
-                l1_ratio=self.l1_ratio, regularization=self.regularization,
-                random_state=self.random_state, verbose=self.verbose,
-                shuffle=self.shuffle)
-
-        self.reconstruction_err_ = _beta_divergence(X, W, H, self.beta_loss,
+            W, H, n_iter = self._fit_transform(X, W=W, H=H)
+
+        self.reconstruction_err_ = _beta_divergence(X, W, H, self._beta_loss,
                                                     square_root=True)
 
         self.n_components_ = H.shape[0]
         self.components_ = H
-        self.n_iter_ = n_iter_
+        self.n_iter_ = n_iter
 
         return W
 
+    def _fit_transform(self, X, y=None, W=None, H=None, update_H=True):
+        """Learn a NMF model for the data X and returns the transformed data.
+
+        This is more efficient than calling fit followed by transform.
```

Comment:
This comment line is already written in the docstring of the public `fit_transform` and is not really relevant here when `update_H=False`.

Maybe, instead state that this private method is used internally both by the public `fit_transform` and `transform` methods, with `update_H=True` for the former and `False` for the latter.